From c6e9300e29a97ae59d7206f562e961266bfe3ed1 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 3 Jul 2024 18:17:57 +0200
Subject: [PATCH 647/841] fix(stm32mp2): correct package types on STM32MP25

Align with datasheet for the macro names, and correct the corresponding
values found in reference manual.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 0ab2336bec6d46e0337bd4ffae557fefc39ee4f2 fix(stm32mp2): correct package types on STM32MP25]
Change-Id: I4d9d2526287d7b6b6df4bbcf1e56540e168388be
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/391282
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     | 6 +++---
 plat/st/stm32mp2/stm32mp2_private.c | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 7fbd07d8a..1af082ccb 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -119,9 +119,9 @@
  ******************************************************************************/
 #if STM32MP25
 #define STM32MP25_PKG_CUSTOM		U(0)
-#define STM32MP25_PKG_AL_TBGA361	U(3)
-#define STM32MP25_PKG_AK_TBGA424	U(4)
-#define STM32MP25_PKG_AI_TBGA436	U(5)
+#define STM32MP25_PKG_AL_VFBGA361	U(1)
+#define STM32MP25_PKG_AK_VFBGA424	U(3)
+#define STM32MP25_PKG_AI_TFBGA436	U(5)
 #define STM32MP25_PKG_UNKNOWN		U(7)
 #endif /* STM32MP25 */
 
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index 398d732ca..b555398b9 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -386,13 +386,13 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 	case STM32MP25_PKG_CUSTOM:
 		pkg = "XX";
 		break;
-	case STM32MP25_PKG_AL_TBGA361:
+	case STM32MP25_PKG_AL_VFBGA361:
 		pkg = "AL";
 		break;
-	case STM32MP25_PKG_AK_TBGA424:
+	case STM32MP25_PKG_AK_VFBGA424:
 		pkg = "AK";
 		break;
-	case STM32MP25_PKG_AI_TBGA436:
+	case STM32MP25_PKG_AI_TFBGA436:
 		pkg = "AI";
 		break;
 #endif /* STM32MP25 */
-- 
2.47.2

